src/nile/core/types/account.py

```diff
@@ -255,6 +268,13 @@ def _get_signer_and_alias(signer, predeployed_info):
     return signer, alias
 
 
+async def _set_estimated_fee_if_none(max_fee, tx):
+    """Estimate max_fee for transaction if max_fee is None."""
+    if max_fee is None:
+        estimated_fee = await tx.estimate_fee()
+        tx.update_fee(estimated_fee)
```

Comment:
I think with this we are good to go, just a small suggestion for organization.

Keeping in mind we are calling this for all the account methods returning tx wrappers. Can't we add this logic in the `BaseTxWrapper` `__post_init__` method to avoid having to call the method repeatedly in account?

Reply:
That was the idea early on, but we'd need `__post_init__` to be async in order to call the underlying `tx.estimate_fee`. It should be possible to do, but I think it's a bit hacky (a la the Account initialization with the async `self.deploy`)

Reply:
Got it, makes sense.